source/Body.h

```diff
@@ -18,21 +18,27 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include "Angle.h"
 #include "Point.h"
+#include "SpriteParameters.h"
 
 #include <cstdint>
+#include <map>
 #include <string>
 
 class DataNode;
 class DataWriter;
 class Government;
 class Mask;
+class Outfit;
 class Sprite;
 
-
-
 // Class representing any object in the game that has a position, velocity, and
 // facing direction and usually also has a sprite.
 class Body {
+public:
+	// States for animation purposes
+	enum BodyState{FLYING, FIRING, LAUNCHING, LANDING, JUMPING, DISABLED, NUM_STATES, CURRENT, TRIGGER};
```

Comment:
I also feel like this all should be more a part of the ship class, because body is also used in a lot of other classes where we dont need this

Reply:
The initial thought behind this was that any body could be animated in different states (e.g. that asteroid mission for New Hope? could have a landing asteroid sprite which glowed red due to friction from atmospheric entry.)

Also, since the core of this development revolved around the selection of sprites, and Body is the sole class afaik responsible for delivering those sprites (from the `GetFrame()` and `GetMask()` functions), I figured that it would be more logical to have something local to Body that determines which sprite to select, rather than Ship.

If I did end up moving these states to the Ship class, there would have to be some interconnection between Ship and Body assuming that Body would continue to deliver the sprites (e.g. Body would have to incorporate states like Ship::FIRING or Ship::JUMPING within its code which I don't think makes much sense. Instead, having Ship incorporate BodyState::FIRING or BodyState::JUMPING seems more clean).

Also to address the concern regarding the other classes which don't utilize these states, keeping the animation state within Body allows for future development in the case that those classes may need to utilize these states (e.g. asteroid example). 

There is not really a justification for the classes which do not, and will not (in the future), need these states, but these classes are handled just as efficiently runtime-wise as before this implementation. They never switch states, so there is no overhead with switching sprites. The only downside will be an increased amount of memory required to display those bodies.

Reply:
Yeah we dont have a memory issue for the game so its fine if all objects are slightly longer